wp/headless-wp/includes/classes/Integrations/Gutenberg.php

```diff
@@ -23,6 +23,124 @@ class Gutenberg {
 	public function register() {
 		add_filter( 'render_block', [ $this, 'render_block' ], 10, 3 );
 		add_filter( 'render_block_core/image', [ $this, 'ensure_image_has_dimensions' ], 9999, 2 );
+		add_action( 'rest_api_init', [ $this, 'extend_content_for_all_post_types' ] );
+	}
+
+	/**
+	 * Get inline block styles.
+	 *
+	 * @param \WP_Post $post    The post.
+	 *
+	 * @return string
+	 */
+	public function get_inline_block_styles( \WP_Post $post ): string {
+		$done   = [];
+		$css = '';
+
+		wp_enqueue_stored_styles();
+		if ( isset( wp_styles()->registered['core-block-supports']->extra['after'] ) ) {
+			$css = end( wp_styles()->registered['core-block-supports']->extra['after'] );
+		}
+
+		$blocks = parse_blocks( $post->post_content );
+
+		return $css . $this->get_blocks_styles( $blocks, $done );
+	}
+
+	/**
+	 * Extend content field for all public post types using REST API filters.
+	 */
+	public function extend_content_for_all_post_types() {
+		$post_types = get_post_types( [ 'public' => true ], 'names' );
+
+		foreach ( $post_types as $post_type ) {
+			add_filter( "rest_prepare_{$post_type}", [ $this, 'extend_post_content' ], 10, 3 );
+		}
+	}
+
+	/**
+	 * Extend the content field with additional data.
+	 *
+	 * @param \WP_REST_Response $data    The response object.
+	 * @param \WP_Post          $post    The post object.
+	 * @param \WP_REST_Request  $request The request object.
+	 *
+	 * @return \WP_REST_Response
+	 */
+	public function extend_post_content( \WP_REST_Response $data, \WP_Post $post, \WP_REST_Request $request ) {
+		// Only extend if content field exists
+		if ( ! isset( $data->data['content'] ) ) {
+			return $data;
+		}
+
+		if ( ! isset( $data->data['content']['rendered'] ) ) {
+			return $data;
+		}
+
+		$params = $request->get_params();
+
+		if ( 'view' !== $params['context'] ) {
+			return $data;
+		}
+
+		if ( ! isset( $params['slug'] ) && ! isset( $params['id'] ) ) {
```

Comment:
@nicholasio In the custom endpoint we are using, we don't have the `slug` and the `id` parameters. Do you think it would be okay if we add a filter here to bypass this condition? Thanks!

Reply:
@phpbits done.